### PR TITLE
Auto-detect RCT1 files location

### PR DESCRIPTION
**Pull request by Gymnasiast**
This needs testing on Windows.

It works on Linux with the RCT1 files downloaded using the Steam console and `download_depot`.

### CODE REVIEW

File: src/openrct2/config/Config.cpp [hunk 1]
```diff
@@ -596,6 +596,49 @@ namespace Config
         }
     }
 
+    /**
+ * Attempts to find the RCT1 installation directory.
+ * @returns Path to RCT2, if found. Empty string otherwise.
```
Comment (Gymnasiast) [inline src/openrct2/config/Config.cpp]: (This comment needs adjusting.)

File: src/openrct2/config/Config.cpp [hunk 2]
```diff
@@ -769,5 +812,13 @@ bool config_find_or_browse_install_directory()
             return false;
         }
     }
+    // While we're at it, also check if the player has RCT1
+    std::string rct1Path = Config::FindRCT1Path();
+    if (!rct1Path.empty())
+    {
+        Memory::Free(gConfigGeneral.rct1_path);
```
Comment (AaronVanGeffen) [inline src/openrct2/config/Config.cpp]: The Memory class is deprecated. Please use `free` instead.

(I really wish we were at a point where we could leave memory management to C++ intrinsics…)

**Comment by Umdlye** [comment]
Works here (Windows 10 x64, Steam version of RCT1). The RCT1 scenarios aren't visible until the second time the game starts though. The other stuff (RCT1 edge textures) appear right away.

**Comment by Gymnasiast** [comment]
I should perhaps trigger a rebuild of the scenario cache if it finds an RCT1 directory,

**Comment by Umdlye** [comment]
It can't find the RCT1 title screen music until the second boot either by the way ;o

File: src/openrct2/config/Config.cpp (hunk 2 above)
Comment (IntelOrca) [reply src/openrct2/config/Config.cpp]: Yes, I didn't get round to removing the remaining ones because there are lots of them.

**Pull request merged**